pkgs/by-name/de/deltachat-desktop/package.nix

```
@@ -64,56 +54,44 @@ buildNpmPackage rec {
   ];
 
   env = {
+    DELTA_CHAT_RPC_SERVER = lib.getExe deltachat-rpc-server;
     ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
-    ESBUILD_BINARY_PATH = "${esbuild'}/bin/esbuild";
-    USE_SYSTEM_LIBDELTACHAT = "true";
-    VERSION_INFO_GIT_REF = src.rev;
+    VERSION_INFO_GIT_REF = finalAttrs.src.rev;
   };
 
-  preBuild = ''
+  buildPhase = ''
+    runHook preBuild
+
     test \
-      $(jq -r '.packages."node_modules/@deltachat/jsonrpc-client".version' package-lock.json) \
+      $(yq -r '.catalogs.default."@deltachat/jsonrpc-client".version' pnpm-lock.yaml) \
       = ${deltachat-rpc-server.version} \
       || (echo "error: deltachat-rpc-server version does not match jsonrpc-client" && exit 1)
 
     test \
-      $(jq -r '.packages."node_modules/electron".version' package-lock.json | grep -E -o "^[0-9]+") \
+      $(yq -r '.importers."packages/target-electron".devDependencies.electron.version' pnpm-lock.yaml | grep -E -o "^[0-9]+") \
       = ${lib.versions.major electron.version} \
       || (echo 'error: electron version doesn not match package-lock.json' && exit 1)
 
-    rm node_modules/@deltachat/stdio-rpc-server-*/deltachat-rpc-server
-    ln -s ${lib.getExe deltachat-rpc-server} node_modules/@deltachat/stdio-rpc-server-linux-*
-  '';
-
-  npmBuildScript = "build4production";
-
-  installPhase = ''
-    runHook preInstall
-
-    npm prune --production
+    #rm node_modules/@deltachat/stdio-rpc-server-*/deltachat-rpc-server
+    #ln -s ${lib.getExe deltachat-rpc-server} node_modules/@deltachat/stdio-rpc-server-linux-*
 
-    mkdir -p $out/lib/node_modules/deltachat-desktop
-    cp -r . $out/lib/node_modules/deltachat-desktop
+    pnpm -w build:electron
 
-    awk '!/^#/ && NF' build/packageignore_list \
-      | xargs -I {} sh -c "rm -rf $out/lib/node_modules/deltachat-desktop/{}" || true
+    pnpm --filter=@deltachat-desktop/target-electron exec electron-builder -- \
```

Your arguments are ignored here because `pnpm` parses arguments differently.
You need to remove the `--`

```suggestion
    pnpm --filter=@deltachat-desktop/target-electron exec electron-builder \
```

This will resolve the error you are facing.

You will find the resulting directory structure in `./packages/target-electron/dist/linux-unpacked/`.